app/src/main/java/chat/rocket/android/members/presentation/MembersPresenter.kt

```diff
@@ -67,4 +92,82 @@ class MembersPresenter @Inject constructor(
             }
         }
     }
+
+    fun toggleOwner(userId: String, isOwner: Boolean = false, notifier: () ->Unit) {
+        launchUI(strategy) {
+            try {
+                if (isOwner)
+                    retryIO(description = "removeOwner($roomId, $roomType, $userId)") { client.removeOwner(roomId, roomType, userId) }
+                else
+                    retryIO(description = "addOwner($roomId, $roomType, $userId)") { client.addOwner(roomId, roomType, userId) }
+                notifier()
+            } catch (ex: RocketChatException) {
+                view.showMessage(ex.message!!) // TODO Remove.
```

Comment:
This is not the way to go. Check how we are handling our exceptions on our codebase please. There are lots of examples demonstrating how to handle it.